src/main/java/io/reactiverse/pgclient/impl/codec/DataType.java

```diff
@@ -79,8 +86,38 @@ public void encodeBinaryInternal(Boolean value, ByteBuf buff) {
       buff.writeBoolean(value);
     }
   };
+  //Commented version cause other datatypes are always the same procedure
+  public static final DataType<boolean[]> BOOL_ARRAY = new DataType<boolean[]>(boolean[].class, 1000) {
 
-  public static DataType<boolean[]> BOOL_ARRAY = new DataType<>(boolean[].class, 1000);
+    @Override
+    public boolean[] decodeBinary(int len, ByteBuf buff) {
+      buff.readerIndex(buff.readerIndex() + 4);//Skip dimensions cause its always 1
+      int offset = buff.readInt();//Read the offset, used to skip the bitmap
+      buff.readerIndex(buff.readerIndex() + 4);//Skip the oid cause we know what type it is
+      int length = buff.readInt(); //Read the length to create a fixed length array
+      boolean[] array = new boolean[length];//Create the array
+      buff.readerIndex(buff.readerIndex() + offset + 4);//Skip, if exists, the bitmap and the lower boundry
```

Comment:
instead use `buff.skipBytes(4)`

Reply:
That wont work, the default would expect the total number of elements in the array. But if the array is empty the header just contains the `0` dimensions, the `0` offset and the `OID`. The check at the start is in my opinion the most effective way to handle 0 size arrays

Reply:
ok the check makes sense then

note: you commented on the `skipBytes` comment which is valid.

Reply:
Oh, sorry did not mean to do that, yeah that part is valid.